code/controllers/Processes/shuttles.dm

```diff
@@ -215,3 +219,116 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 				S.dwidth = M.dwidth
 				S.dheight = M.dheight
 		moveShuttle(M.id, "[M.roundstart_move]", 0)
+
+/datum/transit_allocator
+	var/x1
+	var/x2
+	var/y1
+	var/y2
+	var/z
+	var/list/allocated = list()
+
+/datum/transit_allocator/New()
+	for(var/obj/effect/landmark/L in landmarks_list)
+		if(L.name == "transit_lower")
+			x1 = L.x
+			y1 = L.y
+			z = L.z
+		else if(L.name == "transit_upper")
+			x2 = L.x
+			y2 = L.y
+			z = L.z
+
+// This proc attempts to allocate a transit area, and returns the registered docking port.
+/datum/transit_allocator/proc/allocate(obj/docking_port/mobile/D)
+	var/datum/allocated_transit/A = new(D, src)
+	if(!A.try_to_position())
+		return null
+	for(var/turf/T in block(locate(A.x,A.y,z),locate(A.x+A.w,A.y+A.h,z)))
+		T.ChangeTurf(D.transit_type)
```

Comment:
Yes it does, the mobile port's `transit_type` has the appropriate-direction transit turf type on it.

The mobile port's `dir` has nothing to do with this - It has no connection to the direction the transit turfs go.

If you mean having the correct width and height, that was figured out in the allocated object.